include/libtorrent/aux_/vector.hpp

```diff
@@ -39,11 +39,22 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent { namespace aux {
 
+	template <typename IndexType>
+	struct vector_index_t
```

Comment:
true, I wanted to capture the fact that only integral types are allowed as early as possible, I will change it to this, it will fail with the internal conversion in the `[]` operators.